src/models/db/fulfillments.js

```diff
@@ -39,23 +41,36 @@ function * insertFulfillments (fulfillments, options) {
 
 function * getFulfillment (transferUuid, options) {
   return getTransferId(transferUuid, options).then((transferId) => {
+    if (!transferId) {
+      throw new TransferNotFoundError('This transfer does not exist')
+    }
     return db.selectOne({TRANSFER_ID: transferId},
       options && options.transaction).then((result) => {
-        if (result) {
-          result.transfer_id = transferUuid
+        if (!result) {
+          throw new FulfillmentNotFoundError('This transfer has no fulfillment')
```

Comment:
It might be good to differentiate at least in the error messages between a transfer that is not yet finalized (`This transfer has not yet been fulfilled`) and one that has already expired (`This transfer expired before it was fulfilled`)